lib/opal/cache.rb

```diff
@@ -29,79 +29,147 @@ def fetch(*)
       end
     end
 
-    class FileCache
-      def initialize(dir: nil)
-        @root = File.expand_path '..', Opal.gem_dir
-        @dir = dir || find_dir
-      end
+    unless RUBY_ENGINE == 'opal'
```

Comment:
Thoughts on moving FileCache to its own file and append the `unless` to the `require`?